src/interpret.jl

```diff
@@ -673,6 +735,7 @@ function next_line!(stack, frame, dbstack = nothing)
             pc == nothing && return nothing
         end
         frame.pc[] = pc
+        shouldbreak(frame, pc) && break
```

Comment:
Is something off here?

```julia
> function f(x)
      g(x)
  end
f (generic function with 1 method)

> function g(x)
      print("hello")
      return 2
  end
g (generic function with 1 method)

> breakpoint(g)

> frame = JuliaInterpreter.enter_call(f, 2)
JuliaStackFrame(JuliaInterpreter.JuliaFrameCode(f(x) in Main at REPL[18]:2, CodeInfo(
1 ─ %1 = ($(QuoteNode(g)))(x)
└──      return %1
), Union{Compiled, TypeMapEntry}[#undef, #undef], Tuple{Bool,Function}[#undef, #undef], BitSet([1]), false, false, true), Union{Nothing, Some{Any}}[Some(f), Some(2)], Any[#undef, #undef], Any[], Int64[], Base.RefValue{Any}(nothing), Base.RefValue{JuliaInterpreter.JuliaProgramCounter}(JuliaProgramCounter(1)), Dict(Symbol("#self#")=>1,:x=>2), Any[])

> JuliaInterpreter.next_line!([frame], frame)
ERROR: MethodError: Cannot `convert` an object of type JuliaInterpreter.Breakpoints.Breakpoint to an object of type JuliaInterpreter.JuliaProgramCounter
Closest candidates are:
  convert(::Type{T}, ::T) where T at essentials.jl:154
  JuliaInterpreter.JuliaProgramCounter(::Any) at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\JuliaInterpreter.jl:34
Stacktrace:
 [1] setproperty!(::Base.RefValue{JuliaInterpreter.JuliaProgramCounter}, ::Symbol, ::JuliaInterpreter.Breakpoints.Breakpoint) at .\sysimg.jl:19
 [2] setindex!(::Base.RefValue{JuliaInterpreter.JuliaProgramCounter}, ::JuliaInterpreter.Breakpoints.Breakpoint) at .\refvalue.jl:33
 [3] next_line!(::Array{JuliaStackFrame,1}, ::JuliaStackFrame, ::Nothing) at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\interpret.jl:744
 [4] next_line!(::Array{JuliaStackFrame,1}, ::JuliaStackFrame) at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\interpret.jl:708
 [5] top-level scope at none:0
```

Reply:
```diff
diff --git a/src/interpret.jl b/src/interpret.jl
index 354c9ee..ec4b013 100644
--- a/src/interpret.jl
+++ b/src/interpret.jl
@@ -710,7 +710,7 @@ function next_line!(stack, frame, dbstack = nothing)
         # If this is a goto node, step it and reevaluate
         if isgotonode(expr)
             pc = _step_expr!(stack, frame, pc)
-            pc == nothing && return nothing
+            (pc == nothing || isa(pc, Breakpoint)) && return pc
         elseif dbstack !== nothing && iswrappercall(expr)
             # With splatting it can happen that we do something like ssa = tuple(#self#), _apply(ssa), which
             # confuses the logic here, just step into the first call that's not a builtin
@@ -727,15 +727,15 @@ function next_line!(stack, frame, dbstack = nothing)
                     break
                 else
                     pc = _step_expr!(stack, frame, pc)
-                    pc == nothing && return nothing
+                    (pc == nothing || isa(pc, Breakpoint)) && return pc
                 end
             end
         else
             pc = _step_expr!(stack, frame, pc)
-            pc == nothing && return nothing
+            (pc == nothing || isa(pc, Breakpoint)) && return pc
         end
         frame.pc[] = pc
-        shouldbreak(frame, pc) && break
+        shouldbreak(frame, pc) && return Breakpoint(frame.code, pc)
     end
     maybe_next_call!(stack, frame, pc)
 end
```

maybe?

Reply:
Yes, that looks good. Feel free to push that here.

Reply:
Done